index.js

```diff
@@ -19,16 +19,21 @@ module.exports = {
                 function rule(node) {
                     const source = node.source;
                     if (!source) return;
-                    const value = source.value;
+                    const value = source.value.replace(/\?.*$/, '');
                     if (!value || !value.startsWith('.') || value.endsWith('.js')) return;
 
                     if (!existsSync(resolve(dirname(context.getFilename()), value))) {
+                        let fix
+                        if(!source.value.includes('?')) {
+                            fix = (fixer) => {
+                                return fixer.replaceText(source, `'${value}.js'`);
+                            }
+                        }
+
                         context.report({
                             node,
                             message: 'Relative imports and exports must end with .js',
-                            fix(fixer) {
-                                return fixer.replaceText(source, `'${value}.js'`);
-                            },
+                            fix,
```

Comment:
I made the fix only apply to module paths without `?`. Otherwise it fixed to something like `./foo?raw.js`

Reply:
Would it make sense to fix `./foo?raw` to `./foo.js?raw`? Or does it matter, since the path is going to be handled by Webpack or whatever anyway?

Reply:
Yes, I think it may matter. At least in webpack v4, I think that the "unnecessary" `.js` extension actually breaks a built. v5 and vite are probably smarter but I'm not sure